client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt

```diff
@@ -4,12 +4,109 @@ package org.thechance.common.data.remote.gateway
+                    UserDto.PermissionDto(id = 2, permission = "DRIVER"),
+                    UserDto.PermissionDto(id = 1, permission = "SUPPORT"),
+                    UserDto.PermissionDto(id = 2, permission = "ADMIN")
+                )
+            ),
+            UserDto(
+                id = "f7b087da-8c02-417b-a3db-54c82b5ff5b4",
+                fullName = "asia",
+                username = "asia",
+                email = "asia@example.com",
+                country = "Iraq",
+                permissions = listOf(
+                    UserDto.PermissionDto(id = 3, permission = "END_USER"),
+                    UserDto.PermissionDto(id = 1, permission = "SUPPORT"),
+                    UserDto.PermissionDto(id = 2, permission = "ADMIN"),
+                )
+            ),
+            UserDto(
+                id = "3e1f5d4a-8317-4f13-aa89-2c094652e6a3",
+                fullName = "ali",
+                username = "ali_jamal",
+                email = "ali_jamal@example.com",
+                country = "Iraq",
+                permissions = listOf(
+                    UserDto.PermissionDto(id = 3, permission = "END_USER"),
+                    UserDto.PermissionDto(id = 1, permission = "ADMIN")
+                )
+            ),
+            UserDto(
+                id = "c3d8fe2b-6d36-47ea-964a-57d45e780bce",
+                fullName = "mustafa",
+                username = "mustafa_246",
+                email = "mustafa_246@example.com",
+                country = "Syria",
+                permissions = listOf(
+                    UserDto.PermissionDto(id = 1, permission = "END_USER"),
+                    UserDto.PermissionDto(id = 2, permission = "RESTAURANT")
+                )
+            ),
+            UserDto(
+                id = "7a1bfe39-4b2c-4f76-bde0-82da2eaf9e99",
+                fullName = "sarah ali",
+                username = "sarah_ali_567",
+                email = "sarah_ali_567@example.com",
+                country = "Palestine",
+                permissions = listOf(
+                    UserDto.PermissionDto(id = 3, permission = "END_USER"),
+                    UserDto.PermissionDto(id = 1, permission = "SUPPORT"),
+                    UserDto.PermissionDto(id = 2, permission = "ADMIN"),
+                    UserDto.PermissionDto(id = 3, permission = "DELIVERY")
+                )
+            ),
+            UserDto(
+                id = "8c90c4c6-1e69-47f3-aa59-2edcd6f0057b",
+                fullName = "Jane Davis",
+                username = "jane_davis_890",
+                email = "jane_davis@example.com",
+                country = "Other",
+                permissions = listOf(
+                    UserDto.PermissionDto(id = 1, permission = "END_USER"),
+                    UserDto.PermissionDto(id = 2, permission = "DRIVER")
+                )
+            ),
+        ).toEntity()
+        val startIndex = (page - 1) * numberOfUsers
+        val endIndex = startIndex + numberOfUsers
```

Comment:
endIndex is used in only one place you could remove it and uses it directly in result

Reply:
it makes the code cleaner and more readable